Dev/src/screens/TeacherScreens/ClassTabs/ClassEditScreen.js

```diff
@@ -1,49 +1,142 @@
 import React from "react";
-import { ScrollView, View, StyleSheet, FlatList, TouchableWithoutFeedback, Keyboard, Text, Alert, Share } from "react-native";
+import { ScrollView, View, StyleSheet, FlatList, TouchableWithoutFeedback, Keyboard, Text, Alert, Share, TextInput } from "react-native";
 import StudentCard from "components/StudentCard";
 import colors from "config/colors";
 import studentImages from "config/studentImages";
 import { Icon } from 'react-native-elements';
 import strings from "config/strings";
+import QcActionButton from 'components/QcActionButton';
+import ImageSelectionModal from 'components/ImageSelectionModal';
+import ImageSelectionRow from 'components/ImageSelectionRow';
 import QcParentScreen from "screens/QcParentScreen";
 import FirebaseFunctions from 'config/FirebaseFunctions';
+import LoadingSpinner from 'components/LoadingSpinner';
 
 export class ClassEditScreen extends QcParentScreen {
 
-  state = {
+  componentDidMount() {
 
-    classID: this.props.navigation.state.params.classID,
-    currentClass: this.props.navigation.state.params.currentClass,
-    students: this.props.navigation.state.params.currentClass.students
+    //Sets the screen for firebase analytics
+    FirebaseFunctions.setCurrentScreen("Class Edit Screen", "ClassEditScreen");
 
   }
 
-  componentDidMount() {
+  //Helpers to retrieve the selected image index 
+  getRandomGenderNeutralImage = () => {
+    index = Math.floor(Math.random() * Math.floor(studentImages.genderNeutralImages.length));
+    imageIndex = studentImages.genderNeutralImages[index];
+    return imageIndex;
+  }
 
-    //Sets the screen for firebase analytics
-    FirebaseFunctions.setCurrentScreen("Class Edit Screen", "ClassEditScreen");
+  getRandomMaleImage = () => {
+    index = Math.floor(Math.random() * Math.floor(studentImages.maleImages.length));
+    imageIndex = studentImages.maleImages[index];
+    return imageIndex;
+  }
+
+  getRandomFemaleImage = () => {
+    index = Math.floor(Math.random() * Math.floor(studentImages.femaleImages.length));
+    imageIndex = studentImages.femaleImages[index];

```

Comment:
We need to open an issue to switch from image index to image id. Image index is dangerous since once you delete or insert an image from the middle all the image associations will change.

Reply:
This is fine for now though.